File: code/jvm/im-repository/src/main/kotlin/im/pagination/Pagination.kt
package im.pagination

/**
 * Represents a pagination response.
 * @param total the total number of items
 * @param currentPage the current page
 * @param totalPages the total number of pages
 * @param nextPage the next page number, or null if there is no next page
 * @param prevPage the previous page number, or null if there is no previous page
 */
data class Pagination(
    val total: Long,
    val currentPage: Int,
    val totalPages: Int,
    val nextPage: Int?,
    val prevPage: Int?,
)

File: code/client/web/src/Domain/pagination/Pagination.ts
import { PaginationInfo } from './PaginationInfo';

/**
 * Represents the pagination information for a list of items.
 *
 * @param items The items.
 * @param info The pagination information.
 */
export interface Pagination<T> {
    readonly items: T[];
    readonly info: PaginationInfo;
}
